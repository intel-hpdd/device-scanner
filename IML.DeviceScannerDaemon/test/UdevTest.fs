// Copyright (c) 2018 DDN. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.

module IML.DeviceScannerDaemon.UdevTest

open Udev
open Fable.Import.Jest
open Matchers
open IML.Types.Fixtures
open IML.CommonLibrary
open IML.Types.CommandTypes
open Fable.PowerPack
open Thoth.Json
open IML.Types
open IML.Types.UeventTypes

let private blockDevices = Map.empty

let private snap (x : Result<BlockDevices, exn>) =
    x
    |> Result.unwrap
    |> UeventTypes.BlockDevices.encoder
    |> Encode.encode 2
    |> toMatchSnapshot

test "Adding a new blockdevice" <| fun () ->
    (UdevCommand.Add(fixtures.add))
    |> update blockDevices
    |> snap
test "Changing a blockdevice" <| fun () ->
    let blockDevices' =
        (UdevCommand.Add(fixtures.add))
        |> update blockDevices
        |> Result.unwrap
    (UdevCommand.Change(fixtures.change))
    |> update blockDevices'
    |> snap
test "Removing a blockdevice" <| fun () ->
    let blockDevices' =
        (UdevCommand.Add(fixtures.add))
        |> update blockDevices
        |> Result.unwrap
    (UdevCommand.Remove(fixtures.remove))
    |> update blockDevices'
    |> Result.unwrap
    |> (==) Map.empty
