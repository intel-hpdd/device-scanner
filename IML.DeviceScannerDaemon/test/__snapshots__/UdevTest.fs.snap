// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding a new blockdevice 1`] = `
"{
  \\"/devices/pci0000:00/0000:00:01.1/ata1/host1/target1:0:0/1:0:0:0/block/sdb/sdb1\\": {
    \\"major\\": \\"8\\",
    \\"minor\\": \\"17\\",
    \\"paths\\": [
      \\"/dev/disk/by-id/ata-VBOX_HARDDISK_VB304a0a0f-15e93f07-part1\\",
      \\"/dev/disk/by-path/pci-0000:00:01.1-ata-1.0-part1\\",
      \\"/dev/sdb1\\"
    ],
    \\"devName\\": \\"/dev/sdb1\\",
    \\"devPath\\": \\"/devices/pci0000:00/0000:00:01.1/ata1/host1/target1:0:0/1:0:0:0/block/sdb/sdb1\\",
    \\"devType\\": \\"partition\\",
    \\"idVendor\\": null,
    \\"idModel\\": \\"VBOX_HARDDISK\\",
    \\"idSerial\\": \\"VBOX_HARDDISK_VB304a0a0f-15e93f07\\",
    \\"idFsType\\": \\"LVM2_member\\",
    \\"idFsUsage\\": \\"filesystem\\",
    \\"idFsUuid\\": \\"wCBeLJ-3ors-t60q-f4sB-2Mhc-eLvN-DZ9s9v\\",
    \\"idPartEntryNumber\\": 1,
    \\"size\\": \\"81784832\\",
    \\"scsi80\\": \\"80\\",
    \\"scsi83\\": \\"83\\",
    \\"isReadOnly\\": true,
    \\"dmSlaveMms\\": [],
    \\"dmVgSize\\": null,
    \\"mdDevices\\": [],
    \\"dmMultipathDevicePath\\": true,
    \\"dmLvName\\": \\"swap\\",
    \\"dmVgName\\": \\"centos\\",
    \\"dmUuid\\": \\"LVM-pV8TgNKMJVNrolJgMhVwg4CAeFFAIMC83IU1hvimWWlvmd5xQddtMIqRtjwOuKTz\\",
    \\"mdUuid\\": null
  }
}"
`;

exports[`Changing a blockdevice 1`] = `
"{
  \\"/devices/pci0000:00/0000:00:01.1/ata1/host1/target1:0:0/1:0:0:0/block/sdb/sdb1\\": {
    \\"major\\": \\"8\\",
    \\"minor\\": \\"17\\",
    \\"paths\\": [
      \\"/dev/disk/by-id/ata-VBOX_HARDDISK_VB304a0a0f-15e93f07-part1\\",
      \\"/dev/disk/by-path/pci-0000:00:01.1-ata-1.0-part1\\",
      \\"/dev/sdb1\\"
    ],
    \\"devName\\": \\"/dev/sdb1\\",
    \\"devPath\\": \\"/devices/pci0000:00/0000:00:01.1/ata1/host1/target1:0:0/1:0:0:0/block/sdb/sdb1\\",
    \\"devType\\": \\"partition\\",
    \\"idVendor\\": null,
    \\"idModel\\": \\"VBOX_HARDDISK\\",
    \\"idSerial\\": \\"VBOX_HARDDISK_VB304a0a0f-15e93f07\\",
    \\"idFsType\\": \\"LVM2_member\\",
    \\"idFsUsage\\": \\"filesystem\\",
    \\"idFsUuid\\": \\"wCBeLJ-3ors-t60q-f4sB-2Mhc-eLvN-DZ9s9v\\",
    \\"idPartEntryNumber\\": 1,
    \\"size\\": \\"800000\\",
    \\"scsi80\\": \\"80\\",
    \\"scsi83\\": \\"83\\",
    \\"isReadOnly\\": true,
    \\"dmSlaveMms\\": [],
    \\"dmVgSize\\": null,
    \\"mdDevices\\": [],
    \\"dmMultipathDevicePath\\": true,
    \\"dmLvName\\": \\"swap\\",
    \\"dmVgName\\": \\"centos\\",
    \\"dmUuid\\": \\"LVM-pV8TgNKMJVNrolJgMhVwg4CAeFFAIMC83IU1hvimWWlvmd5xQddtMIqRtjwOuKTz\\",
    \\"mdUuid\\": null
  }
}"
`;
